max_auto_associations_e2e_test.go

```diff
@@ -0,0 +1,598 @@
+//go:build all || e2e
+// +build all e2e
+
+package hedera
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+var initialMetadataList = [][]byte{{1}, {2}, {3}, {4}, {5}, {6}, {7}, {8}, {9}, {10}}
+
+func createNftHelper(t *testing.T, env *IntegrationTestEnv) TokenID {
+	tokenCreate, err := NewTokenCreateTransaction().
+		SetTokenName("Example Collection").SetTokenSymbol("ABC").
+		SetTokenType(TokenTypeNonFungibleUnique).SetDecimals(0).
+		SetInitialSupply(0).SetMaxSupply(10).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).SetSupplyType(TokenSupplyTypeFinite).
+		SetAdminKey(env.Client.GetOperatorPublicKey()).
+		SetFreezeKey(env.Client.GetOperatorPublicKey()).
+		SetSupplyKey(env.Client.GetOperatorPublicKey()).
+		SetMetadataKey(env.Client.GetOperatorPublicKey()).
+		Execute(env.Client)
+
+	require.NoError(t, err)
+	receipt, err := tokenCreate.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+	return *receipt.TokenID
+}
+
+func createFungibleTokenHelper(decimals uint, t *testing.T, env *IntegrationTestEnv) TokenID {
+	tokenCreate, err := NewTokenCreateTransaction().
+		SetNodeAccountIDs(env.NodeAccountIDs).
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetTokenMemo("asdf").
+		SetDecimals(decimals).
+		SetInitialSupply(1_000_000).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetAdminKey(env.Client.GetOperatorPublicKey()).
+		SetFreezeKey(env.Client.GetOperatorPublicKey()).
+		SetWipeKey(env.Client.GetOperatorPublicKey()).
+		SetSupplyKey(env.Client.GetOperatorPublicKey()).
+		SetFreezeDefault(false).
+		Execute(env.Client)
+	require.NoError(t, err)
+
+	receipt, err := tokenCreate.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	return *receipt.TokenID
+
+}
+
+func createAccountHelper(t *testing.T, env *IntegrationTestEnv, maxAutoAssociations int32) (AccountID, PrivateKey) {
```

Comment:
If we are going to have helper functions for future tests(and also taking into consideration past tests). Maybe it would be good to consider creating some reusable utility class holding basic transactions and taking in their parameters. Otherwise it could turn out we have `createAccountHelper` for example in most of the future/past integration test suites. This I guess is a bigger topic and we can bring it up at some point.

Reply:
I am currently thinking about a solution. Will implement it after the HIPs